spring-integration-core/src/main/java/org/springframework/integration/endpoint/SourcePollingChannelAdapter.java

```diff
@@ -206,7 +227,12 @@ protected void handleMessage(Message<?> message) {
 
 	@Override
 	protected Message<?> receiveMessage() {
-		return this.source.receive();
+		if (this.limitingSource != null) {
+			return this.limitingSource.receive(this.maxFetchSize);
+		}
+		else {
+			return this.source.receive();
```

Comment:
Ha! True. "decouple". That's correct word. It isn't poller responsibility to "fetch" the target system.
Although I agree that it would be great to consult "maxFetchSize" from poll to poll. Backpressure ? 😄